src/store/TimeperiodStore.js

```diff
@@ -78,16 +58,36 @@ export default class TimeperiodStore {
                 });
             }
 
-            if (callFromSettings) { setRemain(); }
+            if (callFromSettings) { this.setRemain(); }
 
             if (!this.countdownInterval) {
                 this.countdownInterval = setInterval(() => {
-                    setRemain();
+                    this.setRemain();
                 }, 1000);
```

Comment:
not exactly related to this PR, but now that I do notice it...

`this.remain` should not be an `@obervable`.

Reply:
@brucebinary  I checkout the changes for this file and just remove `@obervable` from `remain`